library/src/main/java/com/bumptech/glide/manager/DefaultConnectivityMonitor.java

```diff
@@ -27,43 +31,97 @@
 
   private boolean isRegistered;
 
-  private final BroadcastReceiver connectivityReceiver =
-      new BroadcastReceiver() {
-        @Override
-        public void onReceive(@NonNull Context context, Intent intent) {
-          boolean wasConnected = isConnected;
-          isConnected = isConnected(context);
-          if (wasConnected != isConnected) {
-            if (Log.isLoggable(TAG, Log.DEBUG)) {
-              Log.d(TAG, "connectivity changed, isConnected: " + isConnected);
-            }
-
-            listener.onConnectivityChanged(isConnected);
-          }
-        }
-      };
+  private final BroadcastReceiver connectivityReceiver;
+  private final ConnectivityManager.NetworkCallback callback;
 
   DefaultConnectivityMonitor(@NonNull Context context, @NonNull ConnectivityListener listener) {
```

Comment:
It's probably cleaner to split this into two classes, a pre M class and a post M class. Then the factory can just pick the appropriate class once, rather than having all of the if/else statements through the single class.

Reply:
I'll check and reformat it